FinalProject/src/com/tms/shop/Application.java

```diff
@@ -0,0 +1,146 @@
+            if (scanner.hasNextInt()) {
+                action = scanner.nextInt();
+            }
+            switch (action) {
+                case 1:
+                    displayProducts();
+                    break;
+                case 2:
+                    addProduct();
+                    break;
+                case 3:
+                    remoteProduct();
+                    break;
+                case 4:
+                    editProduct();
+                    break;
+                case 0:
+                    isTrue = false;
+                    break;
+                default:
+                    System.out.println("Please select an action from the list");
+            }
+        }
+        while (isTrue);
+    }
+
+    private void displayProducts() {
+        List<Product> products = new LinkedList<>();
+        products = shop.getProductList();
+        boolean check = true;
+        int action = 0;
+        do {
+            System.out.println("""
+                    Select to sort the product list
+                    1 - by price (ascending)
+                    2 - by price (decreasing)
+                    3 - by adding (first new, then older)""");
+            if (scanner.hasNextInt()) {
+                action = scanner.nextInt();
+            }
+            switch (action) {
+                case 1:
+                    sortPriceIncrease(products);
+                    System.out.println(shop.toString());
+                    check = false;
+                    break;
+                case 2:
+                    sortPriceDecreasing(products);
+                    System.out.println(shop.toString());
+                    check = false;
+                    break;
+                case 3:
+                    sortAdding(products);
+                    System.out.println(shop.toString());
+                    check = false;
+                    break;
+                default:
+                    System.out.println("Please select a sort from the list");
+            }
+        } while (check);
+
+    }
+
+    private void sortPriceIncrease(List<Product> products) {
+        products.sort(new Comparator<Product>() {
+            @Override
+            public int compare(Product o1, Product o2) {
+                return o1.getPrice() - o2.getPrice();
+            }
+        });
+    }
+
+    private void sortPriceDecreasing(List<Product> products) {
+        products.sort(new Comparator<Product>() {
+            @Override
+            public int compare(Product o1, Product o2) {
+                return o2.getPrice() - o1.getPrice();
+            }
+        });
+    }
+
+    private void sortAdding(List<Product> products) {
+        products.sort(new Comparator<Product>() {
+            @Override
+            public int compare(Product o1, Product o2) {
+                return o2.getIndexProductAdd() - o1.getIndexProductAdd();
+            }
+        });
+    }
+
+    private void addProduct() {
```

Comment:
please don't mix logic of working with products with logic that is working with console. 
refactor this method to 

 private void addProduct(Product product) {
 
 and work with it inside. Same for other methods